### ISSUE TITLE
Can't commit changes to hdf5 backend

### ISSUE THREAD

**Issue by bogovicj** [saalfeldlab/paintera#117]
```
[JavaFX Application Thread] WARN org.janelia.saalfeldlab.paintera.data.mask.MaskedSourceSerializer - Not using relative directory for canvas directory!
[JavaFX Application Thread] WARN org.janelia.saalfeldlab.paintera.data.mask.MaskedSourceSerializer - Not using relative directory for canvas directory!
[JavaFX Application Thread] WARN org.janelia.saalfeldlab.paintera.data.mask.MaskedSourceSerializer - Not using relative directory for canvas directory!
[Thread-410] INFO org.janelia.saalfeldlab.paintera.data.n5.CommitCanvasN5 - Committing canvas
SharedQueue.shutdown() not implemented yet
Exception in thread "Thread-410" ncsa.hdf.hdf5lib.exceptions.HDF5ObjectHeaderException: Object header:Object not found ["H5Omessage.c line 539 in H5O_msg_read_oh(): message type not found"]
	at ch.systemsx.cisd.hdf5.hdf5lib.H5.H5Gopen(Native Method)
	at ch.systemsx.cisd.hdf5.hdf5lib.H5GLO.H5Gopen(H5GLO.java:80)
	at ch.systemsx.cisd.hdf5.HDF5.openGroup(HDF5.java:322)
	at ch.systemsx.cisd.hdf5.HDF5$10.call(HDF5.java:423)
	at ch.systemsx.cisd.hdf5.HDF5$10.call(HDF5.java:1)
	at ch.systemsx.cisd.hdf5.cleanup.CleanUpCallable.call(CleanUpCallable.java:40)
	at ch.systemsx.cisd.hdf5.HDF5.getGroupMembers(HDF5.java:436)
	at ch.systemsx.cisd.hdf5.HDF5BaseReader.getAllGroupMembers(HDF5BaseReader.java:773)
	at ch.systemsx.cisd.hdf5.HDF5BaseReader.getGroupMembers(HDF5BaseReader.java:761)
	at ch.systemsx.cisd.hdf5.HDF5ObjectReadOnlyInfoProviderHandler.getGroupMembers(HDF5ObjectReadOnlyInfoProviderHandler.java:424)
	at org.janelia.saalfeldlab.n5.hdf5.N5HDF5Reader.list(N5HDF5Reader.java:174)
	at org.janelia.saalfeldlab.paintera.N5Helpers.listScaleDatasets(N5Helpers.java:235)
	at org.janelia.saalfeldlab.paintera.N5Helpers.listAndSortScaleDatasets(N5Helpers.java:255)
	at org.janelia.saalfeldlab.paintera.data.n5.CommitCanvasN5.accept(CommitCanvasN5.java:105)
	at org.janelia.saalfeldlab.paintera.data.n5.CommitCanvasN5.accept(CommitCanvasN5.java:58)
	at org.janelia.saalfeldlab.paintera.data.mask.MaskedSource.lambda$persistCanvas$24(MaskedSource.java:455)
	at java.lang.Thread.run(Thread.java:748)

```

**Comment by hanslovsky** [comment]
This will require a specific `CommitCanvas` for each type of `N5Reader` (file system, hdf, etc.).
Alternatively, we could do something like
```java
final String highestResolutionDatasetUniqueLabels = updateLabelToBlockMapping
    ? // current behavior
    : null;
```

**Comment by hanslovsky** [comment]
Just tried to remove the lines and a `LabelMultisetType` check and it does not work yet. It is probably best to dismantle `CommitCanvasN5` and get individual classes for these cases:
 - `IntegerType` or `LabelMultisetType`
   - Single scale data set
   - Multi-resolution data set
   - Paintera dataset
The switch between `IntegerType` and `LabelMultisetType` could be achieved via a member variable

**Comment by hanslovsky** [comment]
Will close this as this should be resolved with #118
